pkg/operator/controllers/subnets/subnet_serviceendpoint.go

```diff
@@ -58,3 +67,7 @@ func (r *reconcileManager) ensureSubnetServiceEndpoints(ctx context.Context, s s
 	}
 	return nil
 }
+
+func gatewayEnabled(cluster *arov1alpha1.Cluster) bool {
+	return len(cluster.Spec.GatewayDomains) > 0
```

Comment:
Should we use `featureProfile` in the cluster CR to check this instead?
Or even both?

Reply:
IMO we should check the featureProfile first, and checking for the list of domains is extra/optional but should also be `>0`.

Reply:
I'm not seeing `featureProfile` in the cluster CR, but to double check that I'm looking at the right thing, the cluster CR is the one you would get with `oc get cluster`, right?

Reply:
@kimorris27 you're right, no featureProfile in the cluster CR, that exists only in the cluster doc 🙁 
I'm cool with just checking the gateway domains if it's the most reliable.

Reply:
my bad, you're correct. What you have is good.